### ISSUE TITLE
Implementing in Tensorflow 2

### ISSUE THREAD

**Issue by karthikm1987** [uber-research/learning-to-reweight-examples#10]
In Tensorflow 2, eager execution seems to remove epsilon (ex_wts_a) from the computational graph, and the derivative of Lg with respect to epsilon is none. Is there a way to make it work in Tensorflow 2?

  epsilon = tf.Variable([0.0 for i in range(batch_size)], dtype=tf.float32)
  with tf.GradientTape(persistent=True) as tape:
    tape.reset()
    yf_hat = damage_classifier_model(Xf)
    binaryCrossEntropy = tf.keras.losses.BinaryCrossentropy()
    loss_f = binaryCrossEntropy(yf,yf_hat)
    Lf = tf.reduce_sum(loss_f * epsilon)
    trainable_vars = damage_classifier_model.trainable_variables
    delta_theta = tape.gradient(Lf, trainable_vars)
    damage_classifier_model.optimizer.apply_gradients(zip(delta_theta, trainable_vars))
    yg_hat = damage_classifier_model(Xg)
    Lg = damage_classifier_model.compiled_loss(yg,yg_hat)

  delta_epsilon = tape.gradient(Lg, epsilon)
  delta_epsilon

**Comment by jneeven** [comment]
Hi, I'm interested in the same issue! Did you find a way to make it work?
